ImprovedAnvil/lib/core/2da.tph

```diff
@@ -357,6 +425,85 @@ COPY_EXISTING ~clastext.2da~ ~override/clastext.2da~
 
   BUT_ONLY
 
+// give additional AC to Sword and Shield
+COPY_EXISTING ~stylbonu.2da~  ~override/stylbonu.2da~
+  COUNT_2DA_COLS ~colcount~
+  SET_2DA_ENTRY 6 5 ~colcount~ ~-1~
+  SET_2DA_ENTRY 6 6 ~colcount~ ~-1~
+  SET_2DA_ENTRY 10 5 ~colcount~ ~-2~
+  SET_2DA_ENTRY 10 6 ~colcount~ ~-2~
+  PRETTY_PRINT_2DA
+
+// add bonuses for *** in Sword and Shield
+COPY_EXISTING ~stylbonu.2da~  ~override/stylbonu.2da~
+  COUNT_2DA_COLS ~colcount~
+  COUNT_2DA_ROWS ~colcount~ ~rowcount~
+  INSERT_2DA_ROW ~rowcount~ ~colcount~ ~SWORDANDSHIELD-3 0                0                0                0                -3               -3               0                0~
+  PRETTY_PRINT_2DA
+
+// update class description for Clerics
+COPY_EXISTING ~clastext.2da~ ~override/clastext.2da~
+  COUNT_2DA_COLS ~colcount~
+  // Pure Cleric class
+  SET_2DA_ENTRY 23 4 ~colcount~ RESOLVE_STR_REF (@12088)
+  // Priest of Talos class
+  SET_2DA_ENTRY 24 4 ~colcount~ RESOLVE_STR_REF (@12126)  
+  // Priest of Helm class
+  SET_2DA_ENTRY 25 4 ~colcount~ RESOLVE_STR_REF (@12127)  
+  // Priest of Lathander class
+  SET_2DA_ENTRY 26 4 ~colcount~ RESOLVE_STR_REF (@12092)
+  // Priest of Tyr class
+  SET_2DA_ENTRY 70 4 ~colcount~ RESOLVE_STR_REF (@12128)
+  // Priest of Tempus class
+  SET_2DA_ENTRY 71 4 ~colcount~ RESOLVE_STR_REF (@12129)
+
+// update kit description for Clerics
+COPY_EXISTING ~kitlist.2da~ ~override/kitlist.2da~
+  COUNT_2DA_COLS ~colcount~
+  // Priest of Talos class
+  SET_2DA_ENTRY 19 4 ~colcount~ RESOLVE_STR_REF (@12126)  
+  // Priest of Helm class
+  SET_2DA_ENTRY 20 4 ~colcount~ RESOLVE_STR_REF (@12127)  
+  // Priest of Lathander class
+  SET_2DA_ENTRY 21 4 ~colcount~ RESOLVE_STR_REF (@12092)
+  // Priest of Tyr class
+  SET_2DA_ENTRY 40 4 ~colcount~ RESOLVE_STR_REF (@12128)
+  // Priest of Tempus class
+  SET_2DA_ENTRY 41 4 ~colcount~ RESOLVE_STR_REF (@12129)
+
 // Add projectiles //
 
-ADD_PROJECTILE ~ImprovedAnvil\resources\projectiles\S!IWD298.PRO~
\ No newline at end of file
+ADD_PROJECTILE ~ImprovedAnvil\resources\projectiles\S!IWD298.PRO~
+ADD_PROJECTILE ~ImprovedAnvil\resources\projectiles\S!FALSED.PRO~
+ADD_PROJECTILE ~ImprovedAnvil\resources\projectiles\S!HARMON.PRO~
+
+// Update conditions for splprot //
```

Comment:
some extra new empty lines would greatly improve readability of this code block

Reply:
Added them, though I think it will be better to create a function, that will add new entries to splprot and create variables, that could be referenced in other files, as current version is very ugly

Reply:
I am not really familiar with this file and how it works to judge that.

Reply:
That's the file, which allows for the most complex and interesting conditions. It's used in opcodes 318/324/326. You can make a spell work or not, based on almost any stat of a creature (resistances, statistics, level, thieving skill etc), you can combine 2 lines to make another condition (str greater than 19 and HP lower than 100, whatever). It has a pretty simple structure and its explained in great detail in IESDP (both for opcode 324 and .2da file).

This file can even check if you have X summons under your control. This is used in this thread to eliminate the problem with Djinni from Wish spell not appearing if you have max number of summons reached:
https://www.gibberlings3.net/forums/topic/30362-whats-the-idea-of-the-summoning-limit/

I use this file and those opcodes extensively in my newest changes and will use it even more, as it gives a great freedom for designing new spells and abilities.

Reply:
Sounds cool.

Reply:
I think that for now we can close this conversation, I will create a function for adding those entries in a cleaner way at some point, but not now.